src/goCover.ts

```diff
@@ -305,3 +309,20 @@ export function toggleCoverageCurrentPackage() {
 		});
 	});
 }
+
+export function isPartOfComment(e: vscode.TextDocumentChangeEvent): boolean {
+	let result = false;
+	e.contentChanges.every(function (change, index) {
+		let line = e.document.lineAt(change.range.start);
+		let text = e.document.getText(line.range);
+		let idx = text.search('//');
+		let changeIdx = change.range.start.character;
+		if (idx === -1 || idx > changeIdx || !change.range.isSingleLine || change.text.includes('\n')) {
```

Comment:
Can you add some tests to cover these cases?

Reply:
Sure!

Reply:
Could you please give me some pointers on how to write tests. I ran the tests and seems like currently there are no tests for code coverage.  I am assuming these would be the steps I would have to follow : 

- Create a go file with code and comments to cover all scenarios. 
- Write some tests for the above file and run them. (use [testCurrentFile](https://github.com/Microsoft/vscode-go/blob/d789de0f9561a1c9aa979a145cb64bcbfbe185cf/src/goTest.ts#L172) function).
- Select some part of the code and delete/add (based on the scenarios in the if condition using [applyUsingTextEditorEdit](https://github.com/Microsoft/vscode-go/blob/d789de0f9561a1c9aa979a145cb64bcbfbe185cf/src/diffUtils.ts#L49) ).
- And check if coverage highlights have been removed or retained. (How do I achieve this?)